tests/functional/src/main/java/org/mule/functional/junit4/TestComponentBuildingDefinitionRegistryFactory.java

```diff
@@ -35,14 +35,19 @@ public class TestComponentBuildingDefinitionRegistryFactory implements Component
   private static final String EXTENSION_SEPARATOR = ",";
 
   private final Map<String, ComponentBuildingDefinitionRegistry> registries = new ConcurrentHashMap<>();
+  private boolean refreshRuntimeComponentBuildingDefinitions = false;
 
   @Override
   public ComponentBuildingDefinitionRegistry create(Set<ExtensionModel> extensionModels) {
     String key = getExtensionsKey(extensionModels);
 
     return registries.computeIfAbsent(key, k -> {
       LOGGER.info(format("Creating new ComponentBuildingDefinitionRegistryFactory for key: %s", key));
-      return new DefaultComponentBuildingDefinitionRegistryFactory().create(extensionModels);
+      DefaultComponentBuildingDefinitionRegistryFactory registryFactory = new DefaultComponentBuildingDefinitionRegistryFactory();
+      if (refreshRuntimeComponentBuildingDefinitions) {
+        registryFactory.refreshRuntimeComponentBuildingDefinitions();
+      }
+      return registryFactory.create(extensionModels);
```

Comment:
`ArtifactFunctionalTestCase#mustRegenerateComponentBuildingDefinitionRegistryFactory` already exists for this

Reply:
It isn't exactly the same, that would create a new `DefaultComponentBuildingDefinitionRegistryFactory` but the `runtimeComponentBuildingDefinitions` would still be the same, I need to recalculate the definitions.